### PR TITLE
docs: versatile replication filters proposal

### PR DESCRIPTION
**Pull request by WilfredAlmeida**
Feature proposal for more versatile replication filters

### CODE REVIEW

**Comment by OrlinVasilev** [comment]
Good stuff @WilfredAlmeida !!

File: proposals/new/versatile-replication-filters.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+Proposal to add more fine-grained replication filters using Regular Expressions. This proposal proposes support for artifacts with [semantically versioned](https://semver.org/) tags.
+
+## Background
+
+As discussed in [issue 8614](https://github.com/goharbor/harbor/issues/8614), the support for more versatile replication filters is been desired by the Harbor user community for a long time now. Users have to frequently change their filter rules every time their images are updated.
+
+## Goals
+
+* Add regex-based filtering for the semantic version tags.
```
Comment (Vad1mo) [inline proposals/new/versatile-replication-filters.md:17]: the reg-ex based filtering should not have only semantic versioning in focus, but artifacts tag in general. 

I would rephrase it to:
* Add regex-based filtering for the artifact tags.

File: proposals/new/versatile-replication-filters.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    
+* Changes made should not break existing installations upon upgrade.
+    
+* Provide seamless migration without needing changes to existing configurations.
+
```
Comment (Vad1mo) [inline proposals/new/versatile-replication-filters.md:22]: I would also see mention of the project name filtering

Bildschirmfoto 2023-04-24 um 13 37 04

**Review by Vad1mo — DISMISSED** [review]

**Comment by WilfredAlmeida** [comment]
I have updated the proposal with discussed approaches.

**Comment by doublest** [comment]
this looks great, i'll vote for the 3rd option. :-)

**Comment by OrlinVasilev** [comment]
voting for 3rd option!

**Comment by WilfredAlmeida** [comment]
Approach 3 for implementation finalized.

**Comment by WilfredAlmeida** [comment]
Draft PR [18723](https://github.com/goharbor/harbor/pull/18723) opened
